Add FILE_SET and RUNTIME_DEPENDENCIES support

solarispika opened the pull request:
## Summary
- Fix `parse_install_targets` to use `parse_install_targets_sub` (was dead code)
- Add FILE_SET formatting support for `target_sources()` and `install(TARGETS)` (CMake 3.23+)
- Add RUNTIME_DEPENDENCIES support:
  - `install(RUNTIME_DEPENDENCY_SET)` as standalone form
  - `RUNTIME_DEPENDENCIES` and `RUNTIME_DEPENDENCY_SET` in `install(TARGETS)` (CMake 3.21+)

## Test plan
- [x] Added test cases for all new features
- [x] All existing tests pass
- [x] Formatted a real project (229 CMakeLists.txt files) without errors

solarispika committed:
Fix parse_install_targets to use parse_install_targets_sub

parse_install_targets_sub was defined but never called. The code
incorrectly used parse_install_targets recursively for designated
kwargs (ARCHIVE, LIBRARY, RUNTIME, etc.), which was overly permissive
since it allowed top-level keywords inside nested groups.

solarispika committed:
Add FILE_SET formatting support (CMake 3.23+)

Add support for the FILE_SET keyword used in target_sources() and
install(TARGETS) commands. FILE_SET requires a set name argument
followed by optional TYPE, BASE_DIRS, FILES, DESTINATION, etc. kwargs.

- Add FileSetNode class for parsing FILE_SET syntax
- Update target_sources() to handle FILE_SET keyword
- Update install(TARGETS) to handle FILE_SET with its specific subparser
- Add tests for adding new keywords

solarispika committed:
Add install RUNTIME_DEPENDENCIES support (CMake 3.21+)

Add support for runtime dependency handling in install():
- RUNTIME_DEPENDENCY_SET as standalone install form
- RUNTIME_DEPENDENCIES and RUNTIME_DEPENDENCY_SET in install(TARGETS)

Both support filtering options (PRE/POST_INCLUDE/EXCLUDE_REGEXES,
POST_INCLUDE/EXCLUDE_FILES, DIRECTORIES).

Spixmaster commented:
Might I ask you to also support `CXX_MODULES_BMI`? It causes issues for me so I think it is also not supported currently.

However, thank you very much for your work! I hope that this project ist not dead.

solarispika committed:
Add CXX_MODULES_BMI support in install(TARGETS) (CMake 3.28+)

CXX_MODULES_BMI is an artifact kind for installing Binary Module
Interface files from C++20 modules. It supports the same options as
other artifact kinds: DESTINATION, PERMISSIONS, CONFIGURATIONS,
COMPONENT, OPTIONAL, EXCLUDE_FROM_ALL.

solarispika commented:
@Spixmaster Sure, I've pushed new commit for CXX_MODULES_BMI support.
Check and see if it helps.
